src/redis-check-rdb.c

```diff
@@ -205,13 +205,15 @@ int redis_check_rdb(char *rdbfilename, FILE *fp) {
     static rio rdb; /* Pointed by global struct riostate. */
     struct stat sb;
 
-    if (isFifo(rdbfilename)) {
-        /* Cannot check RDB over named pipe because fopen blocks until another process opens the FIFO for writing. */
-        return 1;
-    }
-
     int closefile = (fp == NULL);
-    if (fp == NULL && (fp = fopen(rdbfilename,"r")) == NULL) return 1;
+    if (fp == NULL) {
+        if (isFifo(rdbfilename)) {
+            /* Cannot check RDB FIFO because fopen blocks until another process opens the FIFO for writing. */
+            rdbCheckInfo("Cannot check an RDB file that is FIFO: %s", rdbfilename);
+            return 1;
+        }
+        if ((fp = fopen(rdbfilename,"r")) == NULL) return 1;
```

Comment:
maybe we should use this opportunity and improve our error handling, print a message here and include `strerror(errno)`?